Property and history cleanup

stevenger opened the pull request:
# Improved the ssh connection error message from "Failed to run the following command: test -e ~/.jwala" to "SSH connection to <host>:<port> for user <user> failed. Check your host, port, user and password."
    # Removed references to remote.jwala.java.home and PropertyKeys.REMOTE_JAVA_HOME. This should now all be derived from the JVM.
    # Created classes that extend ControlJvmRequest to encapsulate the history messages.
    # Renamed DEPLOY_CONFIG_ARCHIVE to DEPLOY_JVM_ARCHIVE.
    # Added more detail to history message logging for different JVM operations. Still needed for the web server.
    # Removed deprecated @Value("${spring.messaging.topic.serverStates:/topic/server-states}") from JvmControlServiceImpl.
    # Removed validation check for required setenv template file because it's no longer needed. Java home is set when creating the service rather than in the setenv file.
    # Temporarily commenting out the "Expecting ContextRefreshedEvent. Skipping." log message until the logic for this class is added.
    # Refactoring to remove the Aem acronym from AemSsh.
    # Moved ssh properties from AemSshProperty class to PropertyKeys class.
    # Changed DISTRIBUTE_JDK history message to only be written when JDK is distributed rather than every time Jwala checks for a remote JDK. Modified the history text to read Installing JDK  on host "...".
    # Removed unused properties - remote.paths.tomcat.core and jwala.tomcat.service.exe, h2.tcp.server.params, h2.web.server.params, remote.jvm.account.username, remote.jvm.account.password.encrypted.
    # Renamed AemSsh properties to ssh. Removed AemSsh.privateKeyFile=#REQUIRED#, AemSsh.knownHostsFile=#REQUIRED# and AemSsh.port=#REQUIRED" from vars.properties because they're not required.
    # Changed SshConfiguration to require username and password only (removed ssh port). Ssh port now defaults to 22 but can be overridden in vars.properties.

Commit:
# Improved the ssh connection error message from "Failed to run the following command: test -e ~/.jwala" to "SSH connection to <host>:<port> for user <user> failed. Check your host, port, user and password."
# Removed references to remote.jwala.java.home and PropertyKeys.REMOTE_JAVA_HOME. This should now all be derived from the JVM.
# Created classes that extend ControlJvmRequest to encapsulate the history messages.
# Renamed DEPLOY_CONFIG_ARCHIVE to DEPLOY_JVM_ARCHIVE.
# Added more detail to history message logging for different JVM operations. Still needed for the web server.
# Removed deprecated @Value("${spring.messaging.topic.serverStates:/topic/server-states}") from JvmControlServiceImpl.
# Removed validation check for required setenv template file because it's no longer needed. Java home is set when creating the service rather than in the setenv file.
# Temporarily commenting out the "Expecting ContextRefreshedEvent. Skipping." log message until the logic for this class is added.
# Refactoring to remove the Aem acronym from AemSsh.
# Moved ssh properties from AemSshProperty class to PropertyKeys class.
# Changed DISTRIBUTE_JDK history message to only be written when JDK is distributed rather than every time Jwala checks for a remote JDK. Modified the history text to read Installing JDK  on host "...".
# Removed unused properties - remote.paths.tomcat.core and jwala.tomcat.service.exe, h2.tcp.server.params, h2.web.server.params, remote.jvm.account.username, remote.jvm.account.password.encrypted.
# Renamed AemSsh properties to ssh. Removed AemSsh.privateKeyFile=#REQUIRED#, AemSsh.knownHostsFile=#REQUIRED# and AemSsh.port=#REQUIRED" from vars.properties because they're not required.
# Changed SshConfiguration to require username and password only (removed ssh port). Ssh port now defaults to 22 but can be overridden in vars.properties.

Commit:
Removed leftover logging.

Commit:
Merge branch 'master' of https://github.com/cerner/jwala

Commit:
Added missing file SshConfig.java

Commit:
Removed unused property remote.paths.httpd.conf

jmahmoodcc committed:
Merge branch 'master' into property-and-history-cleanup

jmahmoodcc committed:
add heap dump to supported operations in the ControlJvmRequestFratory

jeddcuison commented on jwala-common/src/main/java/com/cerner/jwala/common/request/jvm/StartServiceControlJvmRequest.java:
```diff
@@ -0,0 +1,23 @@
+    public StartServiceControlJvmRequest(final Jvm jvm) {
+        super(jvm.getId(), JvmControlOperation.START);
+        this.jvm = jvm;
+        System.out.println("################## jvm = " + jvm);
```
Remove

jeddcuison replied:
DONE

jeddcuison commented on jwala-services/src/test/java/com/cerner/jwala/service/jvm/impl/JvmServiceImplTest.java:
```diff
@@ -630,9 +615,14 @@ public void testGenerateAndDeployConfig() throws CommandFailureException, IOExce
-        when(Config.mockJvmControlService.controlJvm(eq(new ControlJvmRequest(mockJvm.getId(), JvmControlOperation.DELETE_SERVICE)), any(User.class))).thenReturn(commandOutput);
-        when(Config.mockJvmControlService.controlJvm(eq(new ControlJvmRequest(mockJvm.getId(), JvmControlOperation.DEPLOY_CONFIG_ARCHIVE)), any(User.class))).thenReturn(commandOutput);
-        when(Config.mockJvmControlService.controlJvm(eq(new ControlJvmRequest(mockJvm.getId(), JvmControlOperation.INSTALL_SERVICE)), any(User.class))).thenReturn(commandOutput);
+//        when(Config.mockJvmControlService.controlJvm(eq(new DeleteServiceControlJvmRequest(mockJvm)), any(User.class))).thenReturn(commandOutput);
```
Remove commented out codes

jeddcuison replied:
DONE

jeddcuison committed:
Do clean up

- remove System.out.println
- change employee code to real name
- remove commented out codes
- rearrange code to group similar items
